src/platform/packages/shared/kbn-esql-ast/src/parser/cst_to_ast_converter.ts

```diff
@@ -1466,10 +1466,108 @@ export class CstToAstConverter {
 
   private fromFuseCommand(ctx: cst.FuseCommandContext): ast.ESQLCommand<'fuse'> {
     const command = this.createCommand('fuse', ctx);
+    const fuseTypeCtx = ctx.identifier();
+
+    // FUSE <fuse_method>
+    if (fuseTypeCtx) {
+      const name = this.sanitizeIdentifierString(fuseTypeCtx);
+      const fuseType = Builder.identifier({ name }, this.getParserFields(fuseTypeCtx));
+      command.args.push(fuseType);
+    }
+
+    for (const config of ctx.fuseConfiguration_list()) {
+      this.processFuseConfigurationItem(config, command);
+    }
+
+    const hasIncompleteConfigs = command.args.some((arg) => {
+      return isOptionNode(arg) && arg.incomplete;
+    });
+    if (hasIncompleteConfigs) {
+      command.incomplete = true;
+    }
 
     return command;
   }
 
+  private processFuseConfigurationItem(
+    configCtx: cst.FuseConfigurationContext,
+    command: ast.ESQLCommand<'fuse'>
+  ) {
+    const byContext = configCtx.BY();
+
+    // SCORE BY <score_column>
+    const scoreCtx = configCtx.SCORE();
+    if (scoreCtx && byContext) {
+      const option = this.toOption(scoreCtx.getText().toLowerCase(), configCtx);
+      const scoreColumnCtx = configCtx.qualifiedName();
+
+      if (textExistsAndIsValid(scoreColumnCtx.getText())) {
+        const scoreColumn = this.toColumn(scoreColumnCtx);
+        option.args.push(scoreColumn);
+        option.incomplete = false;
+      } else {
+        option.incomplete = true;
+      }
+
+      command.args.push(option);
+      return;
+    }
+
+    // KEY BY <key_columns>
+    const keyCtx = configCtx.KEY();
+    if (keyCtx && byContext) {
+      const option = this.toOption(keyCtx.getText().toLowerCase(), configCtx);
```

Comment:
The `.getText()` does not have "hidden channel" (whitespace, comments, punctuation), at least spaces will be missing, if there are multiple tokens. In general it is better to cut from source: `this.parser.src.slice(...)`.

But in this instance, I think we should just set it statically:

```suggestion
      const option = this.toOption('key by, configCtx);
```

Also, not ` by`, see other comment on missing " by".